README_EN.md

```diff
@@ -0,0 +1,233 @@
+
+```Swift
+     let controller = RXRViewController(URI: uri)
+     let titleWidget = RXRNavTitleWidget()
+     let alertDialogWidget = RXRAlertDialogWidget()
+     controller.activities = [titleWidget, alertDialogWidget]
+     navigationController?.pushViewController(controller, animated: true)
+```
+
+
+## Customize your own Rexxar Container
+
+First, you inherit `RXRViewController` to implement your own Rexxar Container.
+
+Then, you use the three interfaces provided by Rexxar to make your customization easier.
+
+We will introduce how to use the three interfaces in the following sections.
+
+### Create your own RXRWidget
+
+If you want to implement a native UI feature which can be used by web, for example, displaying a toast or adding pull to refresh UI widget etc, you can inherit `RXRWidget` and implement the three methods: `canPerformWithURL:`, `prepareWithURL:`, `performWithContoller:`.
+
+You can find an example `RXRNavTitleWidget` in Rexxar.
+
+```Objective-C
+@interface RXRNavTitleWidget ()
+
+@property (nonatomic, copy) NSString *title;
+
+@end
+
+
+@implementation RXRNavTitleWidget
+
+- (BOOL)canPerformWithURL:(NSURL *)URL
+{
+  NSString *path = URL.path;
+  if (path && [path isEqualToString:@"/widget/nav_title"]) {
+    return true;
+  }
+  return false;
+}
+
+- (void)prepareWithURL:(NSURL *)URL
+{
+  self.title = [[URL rxr_queryDictionary] rxr_itemForKey:@"title"];
+}
+
+- (void)performWithController:(RXRViewController *)controller
+{
+  if (controller) {
+    controller.title = self.title;
+  }
+}
+
+@end
+```
+
+### Create your own RXRContainerAPI
+
+If you want to offer information computed by native but consumed by web, for example, getting the device's GPS location information, you can create a class conforming to `RXRContainerAPI` protocol and implement the three methods: `shouldInterceptRequest:`, `responseWithRequest:`, `responseData`.
+
+You can find an example `RXRLocContainerAPI` in the RexxarDemo. In this example, `RXRLocContainerAPI` returns the city information. To reduce Demo's complexity, we create it as a mock with false and always the same city information. You can implement your own loc information service on the base of this example.
+
+### Create your own RXRDecorator
+
+If you want to modify the request from Rexxar Container, for example, adding the authentication information in http header, you can inherit `RXRDecorator` and implementing two methods `shouldInterceptRequest:`, `prepareWithRequest:`.
+
+In the RexxarDemo, you will find an example of usage of `RXRRequestDecorator` in `FullRXRViewController` to add the authentication information in http request.
+
+
+## Partial RXRViewcontroller
+
+If you find that a page cann't be implemented totally by Rexxar, you have another option provided by Rexxar: partial RXRViewController. You can use Rexxar to implement a part of the page, and use native code to implement the reset of the page.
```

Comment:
If a page cannot be fully implemented in HTML, you still have a choice to render the page partially with Rexxar. A partial RXRViewController allows you to write part of a page in HTML, and the reset of it in native code.